src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2351,6 +2358,19 @@ void SessionImpl::fileSearchFinished(const TorrentID &id, const Path &savePath,
     }
 }
 
+void SessionImpl::torrentContentDeletionFinished(const QString &torrentName, const QString &errorMessage)
+{
+    if (errorMessage.isEmpty())
+    {
+        LogMsg(tr("Torrent removed. Torrent: \"%1\"").arg(torrentName));
+    }
+    else
+    {
+        LogMsg(tr("Torrent removing done with error(s). Torrent: \"%1\". Error: \"%2\"")
```

Comment:
>done

Is it impossible to fail? If not, something like (not sure if it's correct):
```suggestion
        LogMsg(tr("Torrent removal has encountered error(s). Torrent: \"%1\". Error: \"%2\"")
```
p.s. Can it have multiple errors and if yes shouldn't this also be changed `Error: \"%2\` -> `Error(s): \"%2\`?

Reply:
> Is it impossible to fail?

Well, theoretically this is only possible if you fail to delete the resume data, so that the torrent will be downloaded again next time. But we've never considered this case previously, and I don't remember having any problems with it. At least I'm not going to change that in this PR.

Reply:
> Torrent removal has encountered error(s).

Still, it would be better to emphasize that the errors do not relate to the removal of the torrent itself.

Reply:
> Can it have multiple errors

The files are deleted one by one, and it tries to delete all of them despite the encountered failures, so we may have a number of errors equal to the number of files. But libtorrent provided only one of errors, so I followed its behavior. Usually, if there is a problem with deleting files, then it is the same for all of them, so a single error message is quite enough, I see no point in clogging the log with them.

Reply:
>Still, wouldn't it be better to emphasize that the errors do not relate to the removal of the torrent itself?

Yeah probably, but i've got nothing.

Reply:
Changed it to display messages separately for torrent removal and content removal events:
![003](https://github.com/qbittorrent/qBittorrent/assets/5063477/10c742a6-665e-4142-afac-bee89e78302d)

Reply:
>Changed it to display messages separately for torrent removal and content removal events:

Now it gives 2 log entries for one action... 
Btw i just saw the share limits log messages:
`Removed torrent.`
`Removed torrent and deleted its content.`
Why not re-use them? If you insist to use different strings, you should probably update the share limit strings too.

Reply:
> > Changed it to display messages separately for torrent removal and content removal events:
> 
> Now it gives 2 log entries for one action...

As stated in my comment above, they are two related consecutive events, IMO. So if no one strongly object it I would prefer to handle them separately.


> Btw i just saw the share limits log messages

Share limits log messages are really misleading. They claim that "torrent removed etc." before it's done. They must be replaced by something like "The torrent has reached the share limit, it will be removed along with its content."

Reply:
> Share limits log messages are really misleading. They claim that "torrent removed etc." before it's done. They must be replaced by something like "The torrent has reached the share limit, it will be removed along with its content."

Done.